internal/models/sessions.go

```diff
@@ -34,6 +34,30 @@ func (aal AuthenticatorAssuranceLevel) String() string {
 	}
 }
 
+// CompareAAL returns 0 if both AAL levels are equal, > 0 if A is a higher level than B or < 0 if A is a lower level than B.
+func CompareAAL(a, b AuthenticatorAssuranceLevel) int {
```

Comment:
Just a nit, b int comparisons like this can be error prone, I think we saw it in the crypto package once. It may be worth adding some helper methods:
```Go
func (aal AuthenticatorAssuranceLevel) Above(level AuthenticatorAssuranceLevel) bool
func (aal AuthenticatorAssuranceLevel) Below(level AuthenticatorAssuranceLevel) bool
func (aal AuthenticatorAssuranceLevel) Equal(level AuthenticatorAssuranceLevel) bool
```

Reply:
Yea let's refactor in another PR.